src/models/inbound/consentRequests.model.ts

```diff
@@ -0,0 +1,139 @@
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the Apache License, Version 2.0 (the "License")
+ and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed
+ on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Kevin Leyow - kevin.leyow@modusbox.com
+ --------------
+ ******/
+
+import { BackendRequests } from './backend-requests'
+import {
+  Logger as SDKLogger,
+  ThirdpartyRequests,
+  Errors
+} from '@mojaloop/sdk-standard-components'
+import {
+  thirdparty as tpAPI
+} from '@mojaloop/api-snippets'
+import { HTTPResponseError } from '~/shared/http-response-error'
+import { uuid } from '../../../test/unit/__mocks__/uuidv4';
+
+
+export interface ValidateOTPResponse {
+  isValid: boolean
+}
+
+export interface InboundConsentRequestsRequestModelConfig {
+  logger: SDKLogger.Logger
+  backendRequests: BackendRequests
+  thirdpartyRequests: ThirdpartyRequests
+}
+
+export class InboundConsentRequestsRequestModel {
+  protected config: InboundConsentRequestsRequestModelConfig
+
+  constructor (config: InboundConsentRequestsRequestModelConfig) {
+    this.config = config
+  }
+
+  protected get logger (): SDKLogger.Logger {
+    return this.config.logger
+  }
+
+  protected get backendRequests (): BackendRequests {
+    return this.config.backendRequests
+  }
+
+  protected get thirdpartyRequests (): ThirdpartyRequests {
+    return this.config.thirdpartyRequests
+  }
+
+  async postConsentsRequest (
+    consentRequestsRequestId: string,
+    srcDfspId: string,
+    authToken: string
+  ): Promise<void> {
+    try {
+      const isValidOTP = await this.backendRequests.validateOTPSecret(consentRequestsRequestId, authToken)
+      if (!isValidOTP) {
+        throw new Error('No response returned')
```

Comment:
>maybe instead of message error let use customized Error classes?

I was thinking this would belong in `central-services-error-handling` and that these `Error`'s are just a stop-gap. I didn't want to take in more dependency work since this story is already so big.

> let merge cases No response returned & Invalid OTP in one -> throw InvalidAuthToken (for both cases the result will be the same)

done.